marahgh/assignment3/wordSearch.cpp

```diff
@@ -0,0 +1,212 @@
+#include <iostream>
+#include <vector>
+#include <algorithm>
+using namespace std;
+
+/*A helper function:returns true if the given word
+was found in the given vector, and false otherwise*/
+bool isFoundInVector(vector<string> words, string word) {
+  if (std::find(words.begin(), words.end(), word) != words.end()) {
+    return true;
+  }
+  return false;
+}
+
+/*The grid class,
+contains lower-case letters from whichever alphabet the dictionary uses.*/
+class Grid {
+  char** matrix;
+  int rows;
+  int cols;
+  bool empty;
+
+ public:
+  /*creates an empty grid,
+  a grid with the given dimensions filled by ' ' charachters*/
+  Grid(int rows, int cols)
+      : matrix(nullptr), rows(rows), cols(cols), empty(true) {
+    matrix = new char*[rows];

```

Comment:
There seems to be some duplication between the two constructors.
What you might be able to do is use a delegating constructor:

```c++
  Grid(int rows, int cols) : Grid(vector<char>(rows * cols, ' ')) {}
```